pkg/operator/ceph/object/admin.go

```diff
@@ -149,3 +166,57 @@ func isInvalidFlagError(err error) bool {
 	// it's also returned from some other failures, but this should be rare for Rook
 	return exitCode == 22
 }
+
+func (r *ReconcileCephObjectStore) reconcileRGWAdminOpsUser(cephObjectStore *cephv1.CephObjectStore) error {
+	// Fetch the admin ops user
+	accessKey, secretKey, err := GetAdminOPSUserCredentials(r.context, r.clusterInfo, r.client, cephObjectStore)
+	if err != nil {
+		return errors.Wrap(err, "failed to create or retrieve rgw admin ops user")
+	}
+	r.rgwAdminOpsUserAccessKey = accessKey
+	r.rgwAdminOpsUserSecretKey = secretKey
+
+	return nil
+}
+
+func GetAdminOPSUserCredentials(ctx *clusterd.Context, clusterInfo *cephclient.ClusterInfo, client client.Client, cephObjectStore *cephv1.CephObjectStore) (string, string, error) {
+	if cephObjectStore.Spec.IsExternal() {
+		// Fetch the secret for admin ops user
+		s := &v1.Secret{}
+		err := client.Get(context.TODO(), types.NamespacedName{Name: RGWAdminOpsUserSecretName, Namespace: cephObjectStore.Namespace}, s)
+		if err != nil {
+			return "", "", err
+		}
+
+		if _, ok := s.Data[rgwAdminOpsUserAccessKey]; !ok {
```

Comment:
nit: if you assign it to the local var, you could use it below instead of referencing `s.Data[]` again
```suggestion
		if accessKey, ok := s.Data[rgwAdminOpsUserAccessKey]; !ok {
```